### PR TITLE
INT-3514 fix datacenter user ingestion

### PR DESCRIPTION
**Pull request by mdaum**
# Description

Fix User Ingestion for Jira Software on-prem

## Summary

This change includes a fix to user ingestion for jira software on-prem. It also expands test coverage for on-prem to run against all steps to ensure there are no other runtime errors during ingestion. Also updated documentation for setting up jira software on ARM64 architecture.

## Type of change

Please leave any irrelevant options unchecked.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [X] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [X] My changes properly paginate the target service provider's API
- [X] My changes properly handle rate limiting of the target service provider's
      API
- [X] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [X] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [X] I have updated the `CHANGELOG.md` file to describe my changes
- [X] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/converters/IssueUserRelationshipConverter.ts [hunk 1]
```diff
@@ -8,23 +8,43 @@ import {
   USER_REPORTED_ISSUE_RELATIONSHIP_CLASS,
   USER_REPORTED_ISSUE_RELATIONSHIP_TYPE,
 } from '../entities';
-import { Issue, User } from '../jira';
+import { Issue, User, UserV2, UserV3 } from '../jira';
 import { generateEntityKey } from '../utils';
 
-export function createUserCreatedIssueRelationships(issues: Issue[]) {
+// TODO(mdaum INT-3514): Test changes to this in deployment project
```
Comment (mdaum) [inline src/converters/IssueUserRelationshipConverter.ts:14]: I will remove these todos after I have tested these changes in our managed integration deployment project, which uses this library.

File: src/converters/IssueUserRelationshipConverter.ts [hunk 2]
```diff
@@ -8,23 +8,43 @@ import {
...
-
 function createUserCreatedIssueRelationship(
   user: User,
   issue: Issue,
+  apiVersion: string,
 ): Relationship {
-  const userKey = generateEntityKey(USER_ENTITY_TYPE, user.accountId);
+  let userKey;
+  if (apiVersion === '3') {
+    const castedUser = user as UserV3;
```
Comment (adam-in-ict) [inline src/converters/IssueUserRelationshipConverter.ts:37]: Do you think there's going to be any weirdness with mismatching keys if/when a user switches from v2 to v3 someday?

I don't think that's necessarily an edge case we need to try and handle at the moment, just wondering if that might cause issues and if we'd want to document somewhere as part of this effort.

**Review by adam-in-ict — APPROVED** [review]

File: src/converters/IssueUserRelationshipConverter.ts (hunk 2 above)
Comment (mdaum) [reply src/converters/IssueUserRelationshipConverter.ts:37]: the var names are a bit misleading. V3 api is specifically for Jira Cloud offering. V2 is exclusively for on-prem. So yes migrating from on-prem to cloud and trying to keep your integration instance will result in duplicate users due to the mismatching keys. I think this is an edge case where we recommend they recreate their integration instance if switching from on-prem to cloud, or vice versa.

Comment (mdaum) [reply src/converters/IssueUserRelationshipConverter.ts:37]: adding an aside in the docs on this is a good idea! will add

**Review by adam-in-ict — APPROVED** [review]

**Pull request merged**